### PR TITLE
sysv, armv6/7 fixes

### LINKED ISSUE #265: Unable to force arch for installing exporter
Trying to run node_exporter on an RPI 1. There are releases for it here: https://github.com/prometheus/node_exporter/releases (the armv6 in this case). However i seem unable to force the arch for the module  using hiera `prometheus::node_exporter::arch` since in init.pp it seems to force one of the main arches there. 

[root@dns-pi ~]# cat /proc/device-tree/model
Raspberry Pi Model B Rev 2

## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.8.2 (from raspbian repositories)
- Ruby: ruby 2.3.3p222
- Distribution:   Raspbian GNU/Linux 9.4 (stretch)
- Module version: 6.2

## How to reproduce (e.g Puppet code you use)
classes:
  - prometheus::node_exporter

prometheus::node_exporter::arch: armv6
prometheus::node_exporter::download_url: https://github.com/prometheus/node_exporter/releases/download/v0.16.0/node_exporter-0.16.0.linux-armv6.tar.gz

## What are you seeing
 Evaluation Error: Error while evaluating a Function Call, Unsupported kernel architecture: armv6l at /etc/puppetlabs/code/environments/production/modules/prometheus/manifests/init.pp:189:7 on node dns-pi

## What behaviour did you expect instead
installation of node_exporter

## Any additional information you'd like to impart

Realize it is a pretty fringe case, sorry bout that. Probably simple to add? but not sure how other services would be affected.  I was surprised node_exporter has a few arm releases actually.

### LINKED ISSUE #268: Pupppet sysv fails due to -log.format option
It seems an old option -log.format for prometheus server was removed and prevents the server from starting up with newer prometheus versions (we're using 2.4.3) when using older Sysv init scripts on at least CentOS6. 

This seems to be the affected line and removing -log.format and parameters seem to work. 
https://github.com/voxpupuli/puppet-prometheus/blob/master/templates/prometheus.sysv.erb#L54

## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 4.10
- Ruby: ruby 1.8.7
- Distribution:  CentOS release 6.10 (Final)
- Module version: 6.3.0

## How to reproduce (e.g Puppet code you use)
Use puppet::server class on CentOs6 or sysv class with latest prometheus server version.

## What are you seeing
Prometheus fails to start.

## What behaviour did you expect instead
Prometheus starts and runs.

## Output log
```
[root@statsd.dev /var/log]# service prometheus start
Starting prometheus: 
[root@statsd.dev /var/log]# Error parsing commandline arguments: unknown short flag '-l'
prometheus: error: unknown short flag '-l'
```

## Any additional information you'd like to impart
Looks like this is the issue here. 
https://github.com/voxpupuli/puppet-prometheus/blob/master/templates/prometheus.sysv.erb#L54

Didn't have this issue on systemd startup scripts on Centos7.

### PR DESCRIPTION
**Pull request by defenestration**
#### Pull Request (PR) description
This adds support for armv6/7 releases.
Also on cent6 the 'service_provider' fact gets detected as 'redhat', added fixes to use this like sysv provider. 

#### This Pull Request (PR) fixes the following issues

Fixes #268
Fixes #265

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -134,7 +134,7 @@
 #
 # Requires: see Modulefile
 #
-# Sample Usage:
+# Sample Usage:
```
Comment (bastelfreak) [inline manifests/init.pp]: was there a reason for this change?

**Comment by bastelfreak** [comment]
Hi @defenestration, thanks for the PR! Can you take a look at the failing spec tests?

**Comment by defenestration** [comment]
I added additional commits to hopefully resolve the ruby tests, which seem to just complain about using double quotes in strings without variables.

**Comment by bastelfreak** [comment]
The code looks fine and I'm unable to reproduce the issue locally. I rebased your branch. Let's see if this works now.

**Comment by dhoppe** [comment]
@bastelfreak I think we need to update the fixtures, because the parameter ```-log.format``` has been moved to ```$daemon_flags```. 
https://github.com/voxpupuli/puppet-prometheus/blob/757a5b4d835603c65a7590cc5698bcd0b91a64e4/spec/fixtures/files/prometheus1.sysv#L54
https://github.com/voxpupuli/puppet-prometheus/blob/2a14cf265971fbafeff5454593dbfed9a34cd940/spec/fixtures/files/prometheus2.sysv#L54

**Comment by defenestration** [comment]
I edited the file that the tests use to compare against, removing the -log.format option, which seems to have been the only issue. 

One of the travis builds failed due to apt timing out, but I think it will pass if it is retried, which i can't seem to do.

**Comment by bastelfreak** [comment]
I restarted the failed job

File: manifests/service_reload.pp [hunk 2]
```diff
@@ -10,6 +10,7 @@
       'systemd' => "systemctl reload-or-restart ${init_selector}",
       'upstart' => "service ${init_selector} reload",
       'sysv'    => "/etc/init.d/${init_selector} reload",
+      'redhat'    => "/etc/init.d/${init_selector} reload",
```
[the remark this answers is not in the document]
Comment (bastelfreak) [reply manifests/service_reload.pp]: Please also update the alignment here.

**Comment by wiebe** [comment]
Support for ARMv5, 6 and 7  was already merged, there are double entries now in init.pp!

https://github.com/voxpupuli/puppet-prometheus/blob/492d18f876fa11c051cea4ceee5bf15a9f8e94e0/manifests/init.pp#L186

**Comment by bastelfreak** [comment]
ups. Will provide a patch for this.

**Comment by bastelfreak** [comment]
https://github.com/voxpupuli/puppet-prometheus/pull/295

**Pull request merged**